services/bettersocial/chat/allAnonChat.js

```diff
@@ -0,0 +1,129 @@
+  };
+};
+
+const get_anon_info_from_old_channel = async (oldChannelId, member) => {
+  let anonInfo = {};
+  if (oldChannelId) {
+    let previous_channel = await ChatAnonUserInfo.findOne({
+      where: {
+        channel_id: oldChannelId,
+        my_anon_user_id: member
+      }
+    });
+    anonInfo = {
+      anon_user_info_color_code: previous_channel.anon_user_info_color_code,
+      anon_user_info_color_name: previous_channel.anon_user_info_color_name,
+      anon_user_info_emoji_code: previous_channel.anon_user_info_emoji_code,
+      anon_user_info_emoji_name: previous_channel.anon_user_info_emoji_name
+    };
+  } else {
+    anonInfo = generate_anon_user_info();
+  }
+  return anonInfo;
+};
+
+const handle_anon_to_anon_channel_owner = async (owner, member, context = null) => {
+  // check if channel already exist
+  let channel = await ChatAnonUserInfo.findOne({
+    where: {
+      my_anon_user_id: owner,
+      target_user_id: member,
+      context: context,
+      initiator: owner
+    }
+  });
+  if (channel) {
+    return channel;
+  } else {
+    // handle to create new channel if not exist
+    // generate channel id
+    let channel_id = generate_channel_id_for_anon_chat(owner, member);
+    const emoji = BetterSocialConstantListUtils.getRandomEmoji();
+    const color = BetterSocialConstantListUtils.getRandomColor();
+    let new_channel = await ChatAnonUserInfo.create({
+      channel_id: channel_id,
+      my_anon_user_id: owner,
+      target_user_id: member,
+      anon_user_info_color_code: color.code,
+      anon_user_info_color_name: color.color,
+      anon_user_info_emoji_code: emoji.emoji,
+      anon_user_info_emoji_name: emoji.name,
+      context: context,
+      initiator: owner
+    });
+    return new_channel;
+  }
+};
+
+const handle_anon_to_anon_channel_member = async (
+  channel_owner,
+  member,
+  oldChannelId,
+  context = null
+) => {
+  // check if channel already exist
+  let channel = await ChatAnonUserInfo.findOne({
+    where: {
+      my_anon_user_id: member,
+      target_user_id: channel_owner.my_anon_user_id,
+      context: context,
+      initiator: channel_owner.my_anon_user_id
+    }
+  });
+  if (channel) {
+    return channel;
+  } else {
+    // get anon info from old channel or generate a new one
+    let anonInfo = await get_anon_info_from_old_channel(oldChannelId, member);
+    let new_channel = await ChatAnonUserInfo.create({
+      channel_id: channel_owner.channel_id,
+      my_anon_user_id: member,
+      target_user_id: channel_owner.my_anon_user_id,
+      anon_user_info_color_code: anonInfo.anon_user_info_color_code,
+      anon_user_info_color_name: anonInfo.anon_user_info_color_name,
+      anon_user_info_emoji_code: anonInfo.anon_user_info_emoji_code,
+      anon_user_info_emoji_name: anonInfo.anon_user_info_emoji_name,
+      context: context,
+      initiator: channel_owner.my_anon_user_id
+    });
+    return new_channel;
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The same issue as above applies to the `handle_anon_to_anon_channel_member` function. The emoji and color should be taken from the `anonInfo` object returned by `get_anon_info_from_old_channel`, not generated anew.



<!-- This is an auto-generated comment by OSS CodeRabbit -->